kolibri/content/api.py

```diff
@@ -217,22 +217,34 @@ def filter_popular(self, queryset, name, value):
             return cache.get(cache_key)
 
         # get the most accessed content nodes
+        # search for content nodes that currently exist in the database
         content_counts_sorted = ContentSessionLog.objects \
+            .filter(content_id__in=models.ContentNode.objects.values_list('content_id', flat=True).distinct()) \
             .values_list('content_id', flat=True) \
             .annotate(Count('content_id')) \
             .order_by('-content_id__count')
 
-        most_popular = queryset.filter(content_id__in=list(content_counts_sorted[:10]))
+        most_popular = queryset.filter(content_id__in=list(content_counts_sorted[:20]))
+
+        # remove duplicate content items
+        deduped_list = []
+        content_ids = set()
+        for node in most_popular:
+            if node.content_id not in content_ids:
+                deduped_list.append(node)
+                content_ids.add(node.content_id)
+
+        queryset = most_popular.filter(id__in=[node.id for node in deduped_list])
```

Comment:
is it necessary to do a second query? can you not just return the `deduped_list`

Reply:
the result of this query is returned to another filter method (ex. `filter_by_role`). And the logic in there expects another queryset. If we return a list then we can't do methods on it like `.exclude`.